src/adagrams.js

```diff
@@ -1,15 +1,109 @@
+import { LETTER_POOL, SCORE_CHART } from './constant';
+import { cloneObject } from './helpers';
+
+//Implement drawLetters with weighted randomness
 export const drawLetters = () => {
   // Implement this method for wave 1
+  const availableLetters = cloneObject(LETTER_POOL);
+  const selectedLetters = [];
+
+  for (let i = 0; i < 10; i++) {
+    const total = Object.values(availableLetters).reduce((summary, count) => summary + count, 0);
+    const randLetter = Math.floor(Math.random() * total) + 1;
+
+    let cumulative = 0;
+    for (const [letter, count] of Object.entries(availableLetters)) {
+      cumulative += count;
+      if (randLetter <= cumulative) {
+        selectedLetters.push(letter);
+        availableLetters[letter] -= 1;
+        if (availableLetters[letter] === 0) {
+          delete availableLetters[letter];
+        }
+        break;
+      }
+    }
+  }
+
+  return selectedLetters;
 };
 
+
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  const letterCount = {};
+  input = input.toUpperCase();
+
+  for (const letter of lettersInHand) {
+    if (letterCount[letter]) {
+      letterCount[letter] += 1;
+    } else {
+      letterCount[letter] = 1;
+    }
+  }
+
+  for (const charset of input) {
+    if (letterCount[charset]) {
+      letterCount[charset] -= 1;
+    } else {
+      return false;
```

Comment:
We can make this more concise by following guard clause syntax. Just keep in mind convention (not including the `{}`) of the codebase you are working in. 
```suggestion
    if (!letterCount[charset]) return false;
    letterCount[charset] -= 1;
```